File: generated_code/python/solution_2204_1.py
from typing import List

class Solution:
    def maxSubsequence(self, nums: List[int], k: int) -> List[int]:
        # Pair each element with its original index to preserve order later
        # e.g., nums = [2, 1, 3, 3] -> indexed_nums = [(2, 0), (1, 1), (3, 2), (3, 3)]
        indexed_nums = [(val, idx) for idx, val in enumerate(nums)]
        
        # Sort by value in descending order to identify the k largest elements.
        # Time Complexity for sorting: O(N log N), where N is the length of nums.
        indexed_nums.sort(key=lambda x: x[0], reverse=True)
        
        # Take the top k elements with the largest values
        top_k = indexed_nums[:k]
        
        # Sort these top k elements by their original indices to restore subsequence order
        # Time Complexity for sorting k elements: O(k log k)
        top_k.sort(key=lambda x: x[1])
        
        # Extract the original values to form the final result subsequence
        return [val for val, idx in top_k]

File: generated_code/cpp/solution_2204_1.cpp
#include <vector>
#include <algorithm>
#include <numeric>

using namespace std;

class Solution {
public:
    vector<int> maxSubsequence(vector<int>& nums, int k) {
        int n = nums.size();
        // Create an array of pairs: {element_value, original_index}
        vector<pair<int, int>> indexed_nums(n);
        for (int i = 0; i < n; ++i) {
            indexed_nums[i] = {nums[i], i};
        }

        // Partially sort to find the top k largest elements based on their values.
        // We sort in descending order of values.
        nth_element(indexed_nums.begin(), indexed_nums.begin() + k, indexed_nums.end(), 
                    [](const pair<int, int>& a, const pair<int, int>& b) {
                        return a.first > b.first;
                    });

        // Keep only the top k largest elements
        indexed_nums.resize(k);

        // Sort the chosen k elements based on their original indices to maintain subsequence order
        sort(indexed_nums.begin(), indexed_nums.end(), 
             [](const pair<int, int>& a, const pair<int, int>& b) {
                 return a.second < b.second;
             });

        // Extract the values into the final result array
        vector<int> result(k);
        for (int i = 0; i < k; ++i) {
            result[i] = indexed_nums[i].first;
        }

        return result;
    }
};